### DISCUSSION TITLE
Keystore use with ```cast send```

### DISCUSSION THREAD

**Discussion by SIEcoder** [Cyfrin/foundry-full-course-cu#2543] [Q&A]
Hello everyone. If storing our private keys in a .env file raises security concerns, how can we use ```cast send``` with private key in keystore ?

**Comment by EngrPips** [comment]
Hello @SIEcoder, Here https://book.getfoundry.sh/reference/cli/cast/send. You can find directions on how to use `keystore` wallet when sending transactions.  https://book.getfoundry.sh/reference/cast/cast-wallet-address

<img width="1440" alt="Screenshot 2024-09-25 at 04 41 43" src="https://github.com/user-attachments/assets/be8fc12a-fa62-4973-843d-2c151b998868">

**Reply by SIEcoder** [reply]
Thanks man. Here's what I found to work: cast send [contract address] [function signature] [arguments] [private key]
Sample code snippet 
```cast send 0x59F2f1fCfE2474fD5F0b9BA1E73ca90b143Eb8d0 "store(uint256)" 54321 --keystore ~/.foundry/keystores/My_Key```

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it!

**Reply by SIEcoder** [reply]
Thanks man :)

**Reply by EngrPips** [reply]
Anytime
